pkg/registry/graph.go

```diff
@@ -0,0 +1,295 @@
+package registry
+
+import (
+	"fmt"
+
+	"github.com/openshift/ci-tools/pkg/api"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+// Type declarations
+
+// NodeType identifies the type of registry element a Node refers to
+type NodeType int
+
+const (
+	Workflow NodeType = iota
+	Chain
+	Reference
+)
+
+// Node is an interface that allows a user to identify ancestors and descendants of a step registry element
+type Node interface {
+	Name() string
+	GetNodeType() NodeType
```

Comment:
super nit: this could be `Type()`

Reply:
See the comment on stuttering here https://blog.golang.org/package-names (although this specific case in this PR is not a package stutter, but the concept is the same)